activerecord/test/cases/adapters/sqlite3/transaction_test.rb

```diff
@@ -108,7 +108,7 @@ def shared_cache_flags
     def with_connection(options = {})
       db_config = ActiveRecord::Base.configurations.configs_for(env_name: "arunit", name: "primary")
       conn_options = options.reverse_merge(
-        database: in_memory_db? ? "file::memory:" : db_config.database
+        database: in_memory_db? ? "db/file::memory:" : db_config.database
```

Comment:
Looks like remaining the file is not intended (not expected) to me.

#37986
https://www.sqlite.org/inmemorydb.html

@Edouard-chin Can you confirm about that?

Reply:
`file::memory` is required by sqlite to use `read_uncommited` transaction. Sqlite doesn't allow this transaction level if the db isn't passed as a filepath (thus the need of `file::`)

Reply:
So this change is legit, right?

Reply:
Yep

Reply:
:thinking: I'm a little confused in here. If `db/file::memory:` is passed, does it create file on disk and do not use in-memory one?

Reply:
You are right this is weird. I'll have to double check what's going on. Don't know why that file get created, especially since its size is 0 byte. 

The doc is also pretty clear, but maybe I'm missing something 

> Beginning with SQLite version 3.7.13 (2012-06-11), shared cache can be used on in-memory databases, provided that the database is created using a URI filename. For backwards compatibility, shared cache is always disable for in-memory databases if the unadorned name ":memory:" is used to open the database.

Reply:
This should be resolved by #39158.